pkg/install/billing.go

```diff
@@ -0,0 +1,41 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+)
+
+func (i *Installer) createBillingRecord(ctx context.Context) error {
+	return cosmosdb.RetryOnPreconditionFailed(func() error {
+		var err error
+		_, err = i.billing.Create(ctx, &api.BillingDocument{
+			ID:                        i.doc.ID,
+			Key:                       i.doc.Key,
+			ClusterResourceGroupIDKey: i.doc.ClusterResourceGroupIDKey,
+			Billing: &api.Billing{
+				TenantID: i.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.TenantID,
+				Location: i.doc.OpenShiftCluster.Location,
+			},
+		})
+		// If create return a conflict, this means row is already present in database, updating timestamps
+		if err, ok := err.(*cosmosdb.Error); ok && err.StatusCode == http.StatusConflict {
+			_, err := i.billing.Patch(ctx, i.doc.ID, func(billingdoc *api.BillingDocument) (bool, error) {
+				return false, nil
+			})
+			if err != nil {
+				return err
+			}
+			return nil
+		}
```

Comment:
I don't understand the logic of this.  If the doc exists, we should expect the timestamps to be set.  I don't know why we would want to change them subsequently.

Reply:
Will sync with @shrutir25 if we are ok to not touch the timestamps on creation retry

Reply:
We do not need to touch the timstamp on retry. The billing service will charge only for any VM instances running. So if there aren't any present, it will be a no-op.